drivers/spi/spi_gd32.c

```diff
@@ -29,6 +29,7 @@ LOG_MODULE_REGISTER(spi_gd32);
 
 #if defined(CONFIG_SOC_SERIES_GD32F4XX) || \
 	defined(CONFIG_SOC_SERIES_GD32F403) || \
+	defined(CONFIG_SOC_SERIES_GD32A50X) || \
```

Comment:
Since @gmarull add gd32 clock_control driver for gd32 peripheral, `RCU_APB1EN_OFFSET` is not used anymore. I will removed it latter.

Reply:
I drop these code in c1f38f4c20e7e88351994922d2c19d7e33bba669

Should I squash spi to one commit?